extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -254,19 +253,31 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
       madeInClient
     ) => {
       const enableStudyLazyLoad = false;
-      wadoDicomWebClient.headers = generateWadoHeader();
-      // data is all SOPInstanceUIDs
-      const data = await retrieveStudyMetadata(
-        wadoDicomWebClient,
-        StudyInstanceUID,
-        enableStudyLazyLoad,
-        filters,
-        sortCriteria,
-        sortFunction
-      );
-
-      // first naturalize the data
-      const naturalizedInstancesMetadata = data.map(naturalizeDataset);
+      clientManagerObj.generateWadoHeader();
+
+      let naturalizedInstancesMetadata = [];
+      // search and retrieve in all servers
+      const clients = clientManagerObj.getClients();
+      for (let i = 0; i < clients.length; i++) {
+        const data = await retrieveStudyMetadata(
+          clients[i].wadoDicomWebClient,
+          StudyInstanceUID,
+          enableStudyLazyLoad,
+          filters,
+          sortCriteria,
+          sortFunction
+        );
+        // attach the client Name in each metadata
+        const clientNaturalizedInstancesMetadata = data.map(item => {
+          const converted = naturalizeDataset(item);
+          converted.clientName = clients[i].name;
+          return converted;
+        });
+        // first naturalize the data
+        naturalizedInstancesMetadata = naturalizedInstancesMetadata.concat(
```

Comment:
Do we need to worry about a duplicate series from multiple data sources? Or is it handled?